docs/network-policy.md

```diff
@@ -0,0 +1,365 @@
+
+In some cases we need to explicilty define what sources and sinks a pod is allowed to communicate with, to handle this feature the `spec.ingress` and `spec.egress` fields of a NeworkPolicy can be used 
+
+These sections contain a list of ingress or egress "peers" (the `from` section for `ingress` and the `to` section for `egress`) and a list of IP ports/protocols (the `ports` section) to or from which traffic should be allowed. Each list element of each section is logically OR-ed with other elements.
+
+Each `from`/`to` section can contain the following selectors 
+
+1. `namespaceSelector`: a label selector matching all pods in zero or more Namespaces
+2. `podSelector`: a label selector matching zero or more pods in the same Namespace as the NetworkPolicy
+3. `namespaceSelector` and `podSelector`: when both are present in an element, selects only pods matching the `podSelector` from namespaces matching the `namespaceSelector`
+4. `ipBlock`: an IP network in CIDR notation, with optional exceptions
+
+### `spec.ingress`
+
+Rules defined in `spec.ingress` can match on two main sections, 1.`spec.ingress.from` and 2.`spec.ingress.ports`
+
+1. `spec.ingress.from` 
+
+    Specifies FROM what sources a network policy will allow traffic 
+
+    It contains three selectors which are described further below 
+
+    - `spec.ingress.from.ipBlock` 
+        
+        The ipaddresses from which to allow traffic, contains fields `spec.ingress.from.ipBlock.cidr` to specify which ip address are valid and 
+        `spec.ingress.from.ipBlock.except` to specifiy which address's are not valid 
+    
+    - `spec.ingress.from.namespaceSelector` 
+
+        The Namespaces from which to allow traffic, uses matchLabels to select much like the [`spec.Podselector` field](#**Applying-the-network-policy-to-specific-pods-using-`spec.podSelector`**)
+    
+    - `spec.ingress.from.podSelector` 
+
+        The pods from which to allow traffic, matches the same as described [above](#**Applying-the-network-policy-to-specific-pods-using-`spec.podSelector`**)
+    
+2. `spec.ingress.ports`
+    
+    The ports that the `ingress` rule matches on, contains fields `spec.ingress.ports.port` which can be either numerical or named, if set all port names and numbers will be matched, and `spec.ingress.ports.protocol` matches to the protocol of the provided port 
+
+
+
+### `spec.egress` 
+
+Rules defined in `spec.egress` can match on two main sections, 1.`spec.egress.to` and 2.`spec.egress.ports`
+
+1. `spec.egress.to` 
+
+    specifies TO what sinks a network policy will allow a pod to send traffic 
+
+    It contains three selectors which are described further below 
+
+    - `spec.egress.to.ipBlock` 
+        
+        The ipaddresses which a pod can send traffic to, contains fields `spec.egress.from.ipBlock.cidr` to specify which ip address are valid and 
+        `spec.egress.from.ipBlock.except` to specifiy which address's are not valid 
```

Comment:
you are using the word valid, I am not sure if that is the right word. They are valid IP addresses. Here they specify addresses that the rules apply to